client/src/pages/SignUp.js

```diff
@@ -0,0 +1,243 @@
+import React, { Component } from "react";
+import ReactDOM from "react-dom";
+import {
+  Button,
+  Typography,
+  TextField,
+  Card,
+  CardContent,
+  CardActions,
+  Box,
+} from "@material-ui/core";
+import { withStyles } from "@material-ui/core/styles";
+import BackgroundImg from "../Assets/images/signUpBkg.png";
+import SignIn from "./SignIn";
+
+const signUpPageStyle = (theme) => ({
+  signUpContainer: {
+    fontFamily: '"Montserrat"',
+    margin: theme.spacing(2),
+    height: "100vh",
+  },
+  card: {
+    width: "50%",
+    height: "75%",
+    marginLeft: theme.spacing(20),
+    textAlign: "center",
+  },
+  title: {
+    fontSize: 25,
+    fontWeight: "bold",
+    textAlign: "center",
+  },
+  button: {
+    color: "white",
+    width: theme.spacing(15),
+    background: "#FF510C", //Brighter Orange
+    "&:hover": {
+      backgroundColor: "#FF510C", //Brighter Orange
+    },
+  },
+  titleLogo: {
+    width: "50%",
+    marginLeft: theme.spacing(20),
+    paddingBottom: theme.spacing(2),
+    textAlign: "center",
+  },
+  leftSideContainer: {
+    width: "50%",
+    height: "100vh",
+    float: "left",
+  },
+  rightSideContainer: {
+    width: "50%",
+    height: "100vh",
+    [theme.breakpoints.up("md")]: {
+      backgroundImage: `url(${BackgroundImg})`,
+      backgroundSize: "cover",
+      backgroundPosition: "left",
+    },
+    float: "left",
+  },
+  textField: {
+    margin: theme.spacing(2),
+    width: "30ch",
+  },
+  whiteText: {
+    color: "white",
+  },
+  marginLeft_2: {
+    margin: "2em",
+  },
+});
+
+class SignUpPage extends Component {
+  state = {
+    username: "",
+    email: "",
+    password: "",
+    usernameError: "",
+    emailError: "",
+    passwordError: "",
+    passwordConfirmError: "",
+  };
+
+  validateChange = (e) => {
+    switch (e.target.name) {
+      case "signupName": {
+        if (e.target.value.length === 0) {
+          this.setState({ usernameError: "Field should not be empty." });
+        } else {
+          this.setState({ usernameError: "" });
+          this.setState({ username: e.target.value });
```

Comment:
shouldn't this line be outside of the `if/else`? No matter whether or not something is validated, you should be updating the state because the input value is connected to the state

Reply:
Done.